File: binary-search/Python/binary_search.py
class Solution:
    def search(self, nums: list[int], target: int) -> int:
        low,high = 0, len(nums) - 1
        while (low <= high):
            m = (low + high) // 2
            if target < nums[m]:
                high = m - 1
            elif target > nums[m]:
                low = m + 1
            else:
                return m
        return -1

File: binary-search/C_Cpp/binary_search.c
int search(int* nums, int numsSize, int target) {

    int n = numsSize;
    int low = 0;
    int high = n-1;

    while (low <= high) {
        int m  = (low + high) / 2;
        if (target < nums[m]){
            high = m - 1;
        }
        else if (target > nums[m]){
            low  = m + 1;
        }
        else {
            return m;
        }
    }

    return -1;
}
